node/protocols/data_query/data_query.go

```diff
@@ -83,6 +84,16 @@ func (d *Protocol) PutQueryHistory(key string, val messages.DataQueryRequest) er
 	return nil
 }
 
+// PurgeQueryHistory purges/deletes a query history.
+func (d *Protocol) PurgeQueryHistory(key string) error {
+	if _, ok := d.queryHistory[key]; ok {
```

Comment:
here we can go through all the queries and if they are older than 5 mins we delete them. There is a timestamp attached there so use it to find the difference

Reply:
@filefilegoadmin  done